### ISSUE TITLE
Error while integrating on CakePHP version 1.3

### ISSUE THREAD

**Issue by arpan03** [josegonzalez/cakephp-cake-djjob#7]
I get the following errors when I try to use this plugin with CakePHP v 1.3

```
With app uses
App::uses('ForgotPasswordJob', 'Lib/Job');

Fatal error: Call to undefined method App::uses() in /file.php on line 2
```

```

Without uses with import() and create object

$CakeDjjob = new CakeDjjob;

Fatal error: Class 'CakeDjjob' not found in /file.php on line 31
```

```

With import() direct call

$this->CakeDjjob->enqueue(new ForgotPasswordJob($email));


Notice (music): Undefined property: MyController::$CakeDjjob [APP/controllers/controller.php, line 33]
Fatal error: Call to a member function enqueue() on null in /file.php on line 33
```

Can you help me with this please?

**Comment by josegonzalez** [comment]
You need the 1.3 branch.

**Comment by arpan03** [comment]
Hi Jose,

Thanks for replying back! 

It kind of worked with 1.3 but now i'm facing another issue
`[2016-06-22T14:57:24+00:00] [JOB] attempting to acquire lock for job::38 on host::web.sitename.com pid::20525
PHP Warning:  unserialize(): Function unserialize_jobs() hasn't defined the class it was called for in /PATHTOFILE/DJJob.php on line 461
[2016-06-22T14:57:24+00:00] [JOB] bad handler for job::38
[2016-06-22T14:57:24+00:00] [JOB] failure in job::38
[2016-06-22T14:57:24+00:00] [JOB] attempting to acquire lock for job::38 on host::web.sitename.com pid::20525`

I saw another user mentioning similar issue on this repo. Not sure what the fix was, can you help please?

**Comment by josegonzalez** [comment]
Do not serialize objects or large datasets for your jobs. Can you gist your job code?

**Comment by sandyrewa** [comment]
@josegonzalez 
Hi Jose,

I have created a job 

```
<?php
class ForgotPasswordJob extends CakeJob {
var $email = null;
function __construct($email) {

    $this->email = $email;
 }

function perform() {

    $this->out('Some debug output');
    // Using models
    $this->loadModel('User');
    $user = $this->User->findByEmail($this->email);

    if (empty($user)) {
        return;
    }
    // Complex email stuff here...
    $this->loadComponent('Email');
    $this->Email->_set(array(
        'to' => $this->email,
        'from' => 'noreply@example.com',
        'subject' => 'Forgot your password?',
        'replyTo' => 'noreply@example.com',
        'template' => null,
        'sendAs' => 'both',
    ));
    $this->Email->send("Some text to indicate how to reset password...");
}
}
?>
```

And enueue a job like this 

```
 function forgot_password() {
    if (($email = $this->User->forgot('xyz@abc.com')) != false) {
        App::import('Lib', 'CakeDjjob.ForgotPasswordJob', array(
            'file' => 'jobs' . DS . 'ForgotPasswordJob.php',
        ));

        $this->CakeDjjob->enqueue(new ForgotPasswordJob($email));
    $this->Session->setFlash('You will shortly be sent a password reset email :)');
    } else {
        $this->Session->setFlash('There is no such email in our system. Please try again.');
    }
  }
```

This is working fine and save the job in database in serialized form 

```
O:17:"ForgotPasswordJob":2:{s:5:"email";s:25:"xyx@abc.com";s:10:"_internals";a:0:{}}
```

But when i run the job then we got error which is mentioned above.

**Comment by josegonzalez** [comment]
Your job class needs to be in `app/libs/jobs/forgot_password_job.php` for 1.3.

You should really consider using CakePHP 3 if at all possible. I have a [much better plugin](https://github.com/josegonzalez/cakephp-queuesadilla) for delayed jobs there.

**Comment by sandyrewa** [comment]
@josegonzalez 
Hi Jose,
Thanks for reply
Now it's working fine.
We have 60,000 RSS feeds and we want to run it into batches of 5,000 each in parallel. Is this possible to run it through using single job file but different jobs in parallel. Currently it is running in simultaneously one by one but we want run into parallel.

**Comment by josegonzalez** [comment]
You need to start multiple workers to do process your jobs in parallel. A single worker cannot do multiple things at once.
